From: Pavel Hrdina <phrdina@redhat.com>
Date: Fri, 31 Jul 2015 14:28:00 +0200
Subject: [PATCH virt-manager] details: don't display error if machine is
 missing in XML

Commint 0ddec919 updated the details page.  Now the detail page of
existing domain cannot update the 'machine' value, only prints that
value.  If we cannot get the machine from domain XML, don't pass a None,
but "Unknown" instead.  This can happen if you are connecting with
virt-manager to really old libvirt, the machine value is present in
domain XML since libvirt v0.9.5.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1238981

Signed-off-by: Pavel Hrdina <phrdina@redhat.com>
(cherry picked from commit f81358b02d58b709529a35268cbaf8ce098d803a)
---
 virtManager/details.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/virtManager/details.py b/virtManager/details.py
index 718f50a..8507bd0 100644
--- a/virtManager/details.py
+++ b/virtManager/details.py
@@ -2457,7 +2457,7 @@ class vmmDetails(vmmGObjectUI):
             self.widget("overview-firmware-label").set_text(firmware)
 
         # Machine settings
-        machtype = self.vm.get_machtype()
+        machtype = self.vm.get_machtype() or _("Unknown")
         if self.widget("machine-type").is_visible():
             uiutil.set_list_selection(
                 self.widget("machine-type"), machtype)
